packages/core/src/meta/ObjectMetaField.ts

```diff
@@ -72,16 +72,28 @@ class ObjectMetaFieldInternal<
   }
 
   public override serialize(): ValueOf<T> {
-    return this.transform('serialize');
+    return {
+      ...this.transform('serialize'),
+      ...this.customFields,
+    };
   }
 
   public override clone(): this {
-    return new (<any>this.constructor)(this.name, this.transform('clone'));
+    const cloned = new (<any>this.constructor)(
+      this.name,
+      this.transform('clone'),
+    );
+    cloned.customFields = structuredClone(this.customFields);
```

Comment:
After testing it, it doesn't seem to work as expected.
The initial value of the meta field is set in the constructor, which doesn't have access to `customFields` so the value ends up not including them.
Here's a simple reproduction:
```ts
const field = new ObjectMetaField('example', {
  bool: new BoolMetaField('bool', false),
});
field.set({
  bool: true,
  custom: 'custom',
});

const clone = field.clone();

// This fails:
console.assert(clone.get().custom === 'custom');
```
I think `customFields` should be passed as an optional argument to the constructor instead of being assigned after the object has been created.

Reply:
I modified the constructor originally, but it required more extensive modifications to the call sites to split the custom fields from the built-ins when reading from the scene's meta file.  Instead of assigning to the object's internal `customFields` directly, we can call the `set` method.  This calls `handleChange` properly and only changes a single line. Give the latest commit a try, it works in all my cases.